narwhals/dataframe.py

```diff
@@ -545,7 +547,21 @@ def lazy(self: Self) -> LazyFrame[Any]:
             bar: [[6,7,8]]
             ham: [["a","b","c"]]
         """
-        return self._lazyframe(self._compliant_frame.lazy(), level="lazy")
+        supported_backends = (
+            Implementation.DASK,
+            Implementation.DUCKDB,
+            Implementation.POLARS,
+        )
```

Comment:
RIP Pyspark 😂 Happy to add it later no worries, I am just kidding!

Reply:
Sure, I will try to look into adding PySpark in a follow-up PR